docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -93,6 +93,53 @@ quarkus.http.auth.form.login-page=
 quarkus.http.auth.form.error-page=
 ----
 
+Now that you have disabled redirects for the SPA, you must login and logout programmatically from your client.
+Below are example JavaScript methods for logging into the `j_security_check` endpoint and logging out of the application by destroying the cookie.
+
+[source,javascript]
+----
+const login = () => {
+    // Create an object to represent the form data
+    const formData = new URLSearchParams();
+    formData.append("j_username", username);
+    formData.append("j_password", password);
+
+    // Make an HTTP POST request using fetch against j_security_check endpoint
+    fetch("j_security_check", {
+        method: "POST",
+        body: formData,
+        headers: {
+            "Content-Type": "application/x-www-form-urlencoded",
+        },
+    })
+    .then((response) => {
+        if (response.status === 200) {
+            // Authentication was successful
+            console.log("Authentication successful");
+        } else {
+            // Authentication failed
+            console.error("Invalid credentials");
+        }
+    })
+    .catch((error) => {
+        console.error(error);
+    });
+};
+----
+
+To logout of the SPA you must destroy the cookie and redirect back to your main page.
+
+[source,javascript]
+----
+const logout= () => {
+    // delete the credential cookie essentially killing the session
+    const removeCookie = `quarkus-credential=; Max-Age=0;path=/`;
+    document.cookie = removeCookie;
+    // redirect to main page
+    window.location.href = "/";
```

Comment:
You are documenting logout for SPA with AJAX, I hope this won't confuse users. I think there is many occasions where you want to redirect and it depends on use case whether you want to loose state and use `window.location.href` or keep state and use routing function provided by TypeScript-based SPA framework (for I think it would be challenging to write SPA with only JS + AJAX).

So in short: I think you intention is rather meta-code than something users will copy & paste and your examples are great in that way, but cookie is going to be removed whether you redirect or not. You could just execute some post-logout actions or whatever.

> To logout of the SPA you must destroy the cookie and redirect back to your main page.

That said, I wrote  app that did exactly this redirect in past, so I'm well aware this can be a case. I'll leave decision whether to keep it or not on you, personally I don't like to create an impressions that you need to redirect in SPA, there are other ways to reset state while keeping session data in SPA FWs.

Reply:
@michalvavrik I updated it to say `// perform post logout actions here such as redirecting back to your login page` to leave it up to the user what to do.

Reply:
thanks